### PR TITLE
Add Vitest-style matcher support for test assertions

### PR DESCRIPTION
**Pull request by frostney**
## Summary
- Added new assertion matchers for `expect()`: `toStrictEqual`, `toContainEqual`, `toMatchObject`, and `toMatch`.
- Implemented partial deep equality support in the comparison helpers to power object subset matching.
- Expanded Pascal unit coverage for success, failure, negation, and non-matching type cases.
- Updated docs to describe the new matcher surface in built-ins and testing references.

## Testing
- Added/updated Pascal unit tests in `units/Goccia.Builtins.TestAssertions.Test.pas`.

## Summary by CodeRabbit

* **New Features**
  * Added four new assertion matchers: toStrictEqual (Vitest-compatible alias), toContainEqual, toMatch, and toMatchObject.

* **Documentation**
  * Updated testing docs and matcher reference with examples, compatibility notes, and available matcher highlights.

* **Tests**
  * Expanded unit test coverage to validate the new matchers, negation behavior, subset/cyclic cases, and edge conditions.

### CODE REVIEW

File: units/Goccia.Evaluator.Comparison.pas [hunk 1]
```diff
@@ -350,6 +350,50 @@ function IsDeepEqual(const AActual, AExpected: TGocciaValue): Boolean;
...
+    begin
+      Key := ExpectedKeys[I];
+      if not ActualObj.HasOwnProperty(Key) then
+      begin
+        Result := False;
+        Exit;
+      end;
+
+      if not IsPartialDeepEqual(ActualObj.GetProperty(Key), ExpectedObj.GetProperty(Key)) then
+      begin
```
[the remark this answers is not in the document]
Comment (frostney) [reply units/Goccia.Evaluator.Comparison.pas]: Fixed in 11343f3. Deep and partial comparison now track visited (actual, expected) pairs so cyclic object, set, and map graphs short-circuit safely instead of recursing indefinitely. I also added regression tests covering cyclic toStrictEqual and cyclic toMatchObject.
